src/api/tp.js

```diff
@@ -67,19 +68,34 @@ var TargetProcess = (function() {
     }
 
     var me = this;
-    me.getTask(id).then(success, function (err) {
-        if(err.statusCode === 404) {
-            me.getBug(id).then(success, function (err) {
-                if(err.statusCode === 404) {
-                    done('Task/Bug with Id '+id+' could not be found or access is forbidden.');
-                }
-                else failure(err);
-            });
-        }
-        else failure(err);
+    me.getStory(id).then(function(result) {
+      if (!me.allowLoggingTimeToUserStories) {
```

Comment:
Do we want to do the config check first?  I guess there's no need to make the getStory call if allow is false.